lib/pure/json.nim

```diff
@@ -566,6 +566,11 @@ proc delete*(obj: JsonNode, key: string) =
     raise newException(KeyError, "key not in object")
   obj.fields.del(key)
 
+proc delete*(obj: JsonNode, i: Natural) =
+  ## Deletes ``arr[i]``.
```

Comment:
```suggestion
  ## Deletes ``obj[i]``.
```

This also makes it more consistent with the existing [json.delete](https://github.com/nim-lang/Nim/blob/a81434a890d70e7146c440c59caadc701c6099b3/lib/pure/json.nim#L562..L563)
```Nim
proc delete*(obj: JsonNode, key: string) =
  ## Deletes ``obj[key]``.
```